content/en/blog/_posts/2022-04-15-Increasing-the-security-bar-in-Ingress-NGINX/index.md

```diff
@@ -0,0 +1,127 @@
+
+There are two required changes in your deployments to use this feature:
+* In the image name, add the suffix "-chroot" as the following: gcr.io/k8s-staging-ingress-nginx/controller-chroot:v1.2.0
+* In your manifest, with the capability NET_BIND_SERVICE add the capability SYS_CHROOT so this configuration in the manifest will be like:
+
+```yaml
+capabilities:
+  drop:
+  - ALL
+  add:
+  - NET_BIND_SERVICE
+  - SYS_CHROOT
+```
+
+If you use the Helm Chart, in your `values.yaml`, change the following:
+
+```yaml
+controller:
+  image:
+    chroot: true
+```
+
+** Check with your cluster admin if you can use "SYS_CHROOT" capability before enabling it in your deployment! **
+
+## OK, but how does this increase the security of my Ingress Controller?
+
+Take the following configuration snippet and imagine, for some reason it was added to your nginx.conf:
+```
+location /randomthing/ {
+      alias /;
+      autoindex on;
+}
+```
+
+With this configuration, one can call "http://website.com/randomthing" and get some listing (and access) to the whole filesystem.
+
+Now, can you spot the difference between chrooted and non chrooted Nginx on the listings below?
+
+| not chrooted             | chrooted |
+|--------------------------|------|
+| bin                      | bin  |
+| dev                      | dev  |
+| etc                      | etc  |
+| home                     |      |
+| lib                      | lib  |
+| media                    |      |
+| mnt                      |      |
+| opt                      | opt  |
+| proc                     | proc |
+| root                     |      |
+| run                      | run  |
+| sbin                     |      |
+| srv                      |      |
+| sys                      |      |
+| tmp                      | tmp  |
+| usr                      | usr  |
+| var                      | var  |
+| dbg                      |      |
+| nginx-ingress-controller |      |
+| wait-shutdown            |      |
+
+The one in left side is not chrooted. So NGINX has full access to the filesystem. The one in right side is chrooted, so a new filesystem with only the required files to make NGINX work is created.
+
+## What about other security improvements in this release?
+
+We know that the chroot improvement solves some portion of the risk, but still, someone can try to inject commands to read, for example, nginx.conf file and extract sensitive information.
+
+So, another change in this release (this is opt-out!) is the deep inspector. We know some directives and regexes may be dangerous to NGINX, so the deep inspector will check all fields from an Ingress object (during its reconciliation, and also with admission webhook!) to verify if some field contains these dangerous directives.
+
+We already do this for annotations, and the plan/goal is to move this validation to inside deep inspection in future releases.
```

Comment:
```suggestion
The ingress controller already does this for annotations, and our hope is to move this existing validation to happen inside
deep inspection as part of a future release.

Reply:
I've changed the word hope to goal. Hope seems to much dramatic (in the sense of "we hope someday this will happen" instead of "we plan this to real happen")